src/DotNetWorker/Invocation/DefaultFunctionExecutor.cs

```diff
@@ -1,20 +1,93 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System;
 using System.Linq;
 using System.Threading.Tasks;
-using Microsoft.Azure.Functions.Worker.Pipeline;
 
 namespace Microsoft.Azure.Functions.Worker.Invocation
 {
     internal class DefaultFunctionExecutor : IFunctionExecutor
     {
+        private const string FunctionReturnType = "FunctionsWorkerReturnReserved";
+        private static readonly Type HttpType = typeof(HttpResponseData);
+
         public async Task ExecuteAsync(FunctionContext context)
         {
             var invoker = context.FunctionDefinition.Invoker;
             object? instance = invoker.CreateInstance(context.InstanceServices);
             object? result = await invoker.InvokeAsync(instance, context.FunctionDefinition.Parameters.Select(p => p.Value).ToArray());
-            context.InvocationResult = result;
+
+            object? returnVal = null;
+            if (result is not null)
+            {
+                object? returnFromType = AddOutputBindingsAndGetReturn(context, result);
+                returnVal = returnFromType ?? result;
+            }
+
+            context.InvocationResult = returnVal;
+        }
+
+        internal static object? AddOutputBindingsAndGetReturn(FunctionContext context, object result)
+        {
+            object? returnVal = null;
+            Type resultType = result.GetType();
+
+            if (HttpType.IsAssignableFrom(resultType))
+            {
```

Comment:
How much of this calculating can we push to indexing time? That'd let us do it once at startup and not every execution.

Reply:
... and add any new stuff to FunctionMetadata so we can quickly get the data to the right place here.

Reply:
That's a great point! Yeah, we can probably pre-compute information such as -- which properties have bindings, whether it's a method binding, and even the HttpType check all in the metadata.

I can take that item separately, or update this PR, based on how rest of the review looks. But I think it won't affect the programming model, so maybe we don't need to block this PR on it, but definitely follow-up on it ASAP (this release or next).

Reply:
I moved some stuff to indexing, and some of these type checks aren't required anymore. There's a few additional things that Fabio highlighted we could move to indexing from here later. So I will leave that portion for a future PR.